docs/contributing-ci.md

```diff
@@ -1,3 +1,84 @@
-# CI
+# Continuous Integration (CI)
 
-... more coming soon
+The Wasmtime and Cranelift projects heavily rely on Continuous Integration (CI)
+to ensure everything keeps working and keep the final end state of the code at
+consistently high quality. The CI setup for this repository is relatively
+involved and extensive, and so it's worth covering here how it's organized and
+what's expected of contributors.
+
+All CI currently happens on GitHub Actions and is configured in the [`.github`
+directory][dir] of the repository.
+
+[dir]: https://github.com/bytecodealliance/wasmtime/tree/master/.github
+
+## PRs and CI
+
+Currently the full CI test suite runs on every Pull Request. All PRs need to
+have that lovely green checkmark before being candidates for being merged. If a
+test is failing you'll want to check out the logs on CI and fix it before the PR
+can be merged.
+
+PR authors are expected to fix CI failures in their PR, unless the CI failure is
+systemic and unrelated to the PR. In that case other maintainers should be
+alerted to ensure that the problem can be addressed.
+
+## Tests run on CI
+
+While this may not be fully exhaustive, the general idea of all the checks we
+run on CI looks like this:
+
+* Code formatting - we run `cargo fmt -- --check` on CI to ensure that all code
+  in the repository is formatted with rustfmt. All PRs are expected to be
+  formatted with the latest stable version of rustfmt.
+
+* Book documentation tests - code snippets (Rust ones at least) in the book
+  documentation ([the `docs`
+  folder](https://github.com/bytecodealliance/wasmtime/tree/master/docs)) are
+  tested on CI to ensure they are working.
+
+* Crate tests - the moral equivalent of `cargo test --all` and `cargo test --all
+  --release` is executed on CI. This means that all workspace crates have their
+  entire test suite run, documentation tests and all, in both debug and release
+  mode. Additionally we execute all crate tests on macOS, Windows, and Linux, to
+  ensure that everything works on all the platforms.
+
+* Fuzz regression tests - we take a random sampling of the [fuzz
+  corpus](https://github.com/bytecodealliance/wasmtime-libfuzzer-corpus) and run
+  it through the fuzzers. This is mostly intended to be a pretty quick
+  regression test, most of our fuzzing happens on
+  [oss-fuzz](https://oss-fuzz.com).
```

Comment:
FWIW, it also makes sure that the fuzz targets themselves don't stop compiling.

A bit of an aside, but we were wondering whether it makes sense to keep testing a sampling of each target's corpus, and I think the answer is "yes" because running the actual samples takes very little time, and we want to keep building the fuzz targets in CI anyways (which is what takes most of the CI time) to make sure that the targets themselves don't break.